src/app.js

```diff
@@ -1,3 +1,36 @@
 'use strict';
 
-// Write your code here
+const { generateRandomNumber } = require('./modules/generateRandomNumber');
+const { checkIsValidUserInput } = require('./modules/checkIsValidUserInput');
+const { getBullsAndCows } = require('./modules/getBullsAndCows');
+
+const readline = require('node:readline');
+
+const bcGame = () => {
+  const generatedNum = generateRandomNumber();
+
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  const askUser = () => {
+    rl.question('Write a 4 digit number: ', (answer) => {
+      if (checkIsValidUserInput(answer)) {
+        const result = getBullsAndCows(+answer, generatedNum);
+
+        if (result.bulls === 4) {
+          rl.close();
+        } else {
+          askUser();
+        }
+      } else {
+        askUser();
+      }
+    });
+  };
```

Comment:
This is too overcomplicated logically. You need one statement where you ask the user and one to close the prompt. Writing spaghetti if else's is not a good solution.
```
if([all conditions that should trigger new prompt]) {

}
rl.close()
```